src/NUnitFramework/framework/Internal/Execution/WorkItem.cs

```diff
@@ -230,87 +230,66 @@ public virtual void Execute()
                 timeout = (int)Test.Properties.Get(PropertyNames.Timeout);
 
 #if SILVERLIGHT || NETCF
-            if (Test.RequiresThread || Test is TestMethod && timeout > 0)
+            if (Test is TestMethod)
                 RunTestOnOwnThread(timeout);
             else
                 RunTest();
 #elif PORTABLE
             RunTest();
 #else
-            ApartmentState currentApartment = Thread.CurrentThread.GetApartmentState();
+            currentApartment = Thread.CurrentThread.GetApartmentState();
 
-            if (Test.RequiresThread || Test is TestMethod && timeout > 0 || currentApartment != TargetApartment && TargetApartment != ApartmentState.Unknown)
+            if (Test is TestMethod)
```

Comment:
There will still need to be threads for tests that require their own threads, have a required apartment setting, or have timeouts.  I do not see any way around that.

Always running the test method on its own thread provides consistency so that terminating a thread is always done in the same place, and works the same, no matter what the reason (abort or timeout).  
It also allows everything to clean up after a test the same way, whether it was aborted, stopped, timeout, or just ended normally.

Not sure what you mean about STA being placed on a class under the SimpleWorkItemDispatcher?  I may have missed something, but I thought I had the handled :(